### PR TITLE
[SelectionDAG] Add and use SDNode::getAsAPIntVal() helper

### PR DESCRIPTION
**Pull request by asb**
This is the logical equivalent for #76710 for APInt and uses the same naming scheme.

Converted existing users through:
`git grep -l "cast<ConstantSDNode>\(.*\).*getAPIntValueValue" | xargs sed -E -i 's/cast<ConstantSDNode>\((.*)\)->getAPIntValue/\1->getAsAPIntVal/'`

### CODE REVIEW

File: llvm/include/llvm/CodeGen/SelectionDAGNodes.h [hunk 1]
```diff
@@ -932,6 +932,9 @@ END_TWO_BYTE_PACK()
   /// Helper method returns the APInt of a ConstantSDNode operand.
   inline const APInt &getConstantOperandAPInt(unsigned Num) const;
 
+  /// Helper method returns the APInt value of a ConstantSDNode.
+  inline const APInt &getAsAPIntVal() const;
+
```
Comment (KanRobert) [inline llvm/include/llvm/CodeGen/SelectionDAGNodes.h:937]: Why not directly put the definition here?

Comment (asb) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h:937]: The definition of ConstantSDNode doesn't come until later in the file, and this matches the approach used by the neighbouring getters.

**Review by KanRobert — APPROVED** [review]
LGTM

**Review by RKSimon — APPROVED** [review]
LGTM

**Comment by RKSimon** [comment]
There might be a few cases that would benefit from the getAsAPFloat equivalent as well

**Pull request merged**